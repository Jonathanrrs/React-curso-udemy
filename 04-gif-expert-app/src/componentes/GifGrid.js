import React, { useState, useEffect } from 'react'

export const GifGrid = ({category}) => {

    const [count, setCount] = useState(0);

    useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async() => {
        const url = `https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=vD2873pCOTBaVMmRB83BYa1KkLi62PhF`
        const resp = await fetch(url);
        const {data} = await resp.json();
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gifs);
    }

    return (
        <div>
            <h3>{category}</h3>
            <h1>{count}</h1>
            <button onClick={() => setCount(count+1)}></button>
        </div>
    )
}
