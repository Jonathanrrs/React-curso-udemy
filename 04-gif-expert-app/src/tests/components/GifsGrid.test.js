import React from 'react';
import {shallow} from 'enzyme';
import { GifsGrid } from '../../componentes/GifsGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Probar el componente GifsGrid', () => {
    const category = 'Naruto';

    test('Debe mostrar el componente correctamente ', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(< GifsGrid category={category} />);
        
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar items cuando se cargan imÃ¡genenes en useFetchGifs ', () => {
        /* debemos realizar un mock, que es como fingir algo que pasa */
        const gifs = [
            {
                id: 'Abc',
                url: 'https://localhost/cualquier/cosa.jpg',
                title: 'Cualquier cosa'
            }
        ]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(< GifsGrid category={category} />);

        expect(wrapper).toMatchSnapshot();

    });
    
    
})
