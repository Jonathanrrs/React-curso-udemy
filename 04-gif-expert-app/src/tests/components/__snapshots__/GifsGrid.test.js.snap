// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar el componente GifsGrid Debe mostrar el componente correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Naruto
  </h3>
  <p
    className="card animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Probar el componente GifsGrid Debe mostrar items cuando se cargan im√°genenes en useFetchGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Naruto
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="Abc"
      key="Abc"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
